src/Http/Headers/src/HeaderNames.cs

```diff
@@ -274,10 +277,16 @@ public static class HeaderNames
         /// <summary>Gets the <c>WWW-Authenticate</c> HTTP header name.</summary>
         public static readonly string WWWAuthenticate = "WWW-Authenticate";
 
+        /// <summary>Gets the <c>X-Content-Type-Options</c> HTTP header name.</summary>
+        public static readonly string XContentTypeOptions = "X-Content-Type-Options";
```

Comment:
> Why are you adding Link, X-Content-Type-Options and X-XSS-Protection here?

[www.webpagetest.org](https://www.webpagetest.org/) is one of the commonly used performance testing sites (along with Lighthouse based things for core web vitals e.g. [web.dev/measure](https://web.dev/measure/)); and it will call you out; rightly or wrongly, for not having `X-Content-Type-Options` and `X-XSS-Protection` set; amongst others:

Funnily `X-Content-Type-Options` is the only one [dotnet.microsoft.com](https://dotnet.microsoft.com/) does set:

![image](https://user-images.githubusercontent.com/1142958/112914232-5e640d00-90f3-11eb-9a49-dc4524780817.png)

![image](https://user-images.githubusercontent.com/1142958/112914441-e21df980-90f3-11eb-90ba-5a191ec753d7.png)

vs

![image](https://user-images.githubusercontent.com/1142958/112914547-2f01d000-90f4-11eb-99bc-2df4fee827da.png)

![image](https://user-images.githubusercontent.com/1142958/112914615-5062bc00-90f4-11eb-91f6-627043a47d27.png)

Just looks bad if you are explicitly called out for not setting them; but the Framework doesn't support them as a pre-defined headername when it has 90 other ones...

The `Link` header [rfc5988](https://tools.ietf.org/html/rfc5988); is a good one for using for preloading of fonts prior to the css being parsed and html starting to render to increase webpage performance and reduce [flash of unstyled content (FOUC)](https://en.wikipedia.org/wiki/Flash_of_unstyled_content) or flash of invisible text (FOIT) if you mark the text as unrenderable until the font is downloaded.

e.g. Adding this header on the landing HTML page

`link: <https://assets.ageofascent.net/fonts/teko-light-latin.woff2>; rel=preload; as=font; crossorigin=anonymous,<https://assets.ageofascent.net/fonts/pt_sans-web-regular.woff>; rel=preload; as=font; crossorigin=anonymous`

Chrome/Edge will then immediately start downloading the fonts; without yet having encountered the fonts in the parsing of the HTML (triggered by evaluating its css)

![image](https://user-images.githubusercontent.com/1142958/112915913-6756dd80-90f7-11eb-8e76-4ed5e834faa7.png)

The browsers will also tell you off if you then don't use the preloaded font within the first few seconds of page load:

![image](https://user-images.githubusercontent.com/1142958/112916498-93269300-90f8-11eb-9271-f8b9fffe15bd.png)

Anyway `Link` is one of the missed headers I use; so fringe benefit in adding for me with this change 😉

Its not adding it to the pre-reserved Kestrel headers; just to the `HeaderNames` constants; as otherwise they look fugly in use:

![image](https://user-images.githubusercontent.com/1142958/112916872-514a1c80-90f9-11eb-8c3e-fc03c046abc1.png)

Though since you made me look; seems should add `"Referrer-Policy"` also 🤔